### PR TITLE
Fixes approach definition for Std 229 fluid cooler reports and adds cooling tower implementation

### PR DESCRIPTION
**Pull request by jcyuan2020**
Pull request overview
---------------------
 - Fixes an issue in Std 229 report that the `approach` of a cooling tower was defined incorrectly;
 - Adds Std 229 reports to cooling towers as well, which are similar to fluid cooler implementations;
 
**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

**Comment by nrel-bot-2c** [comment]
@jcyuan2020 @myoldmopar it has been 28 days since this pull request was last updated.

File: src/EnergyPlus/FluidCoolers.cc [hunk 1]
```diff
@@ -2002,7 +2002,7 @@ void FluidCoolerspecs::update(EnergyPlusData &state)
 
     // Check if OutletWaterTemp is below the minimum condenser loop temp and warn user
     Real64 LoopMinTemp = state.dataPlnt->PlantLoop(this->plantLoc.loopNum).MinTemp;
-    if (this->OutletWaterTemp < LoopMinTemp && this->WaterMassFlowRate > 0.0) {
+    if (this->OutletWaterTemp<LoopMinTemp &&this->WaterMassFlowRate> 0.0) {
```
Comment (Myoldmopar) [inline src/EnergyPlus/FluidCoolers.cc]: Clang format comment: Reverse the order or create an intermediate variable

Comment (mjwitte) [reply src/EnergyPlus/FluidCoolers.cc]: Or just put the code back where it was and don't let VS reformat after that.

Comment (jcyuan2020) [reply src/EnergyPlus/FluidCoolers.cc]: Thanks @mjwitte and @Myoldmopar . The clang formatting worked ok after putting brackets to the two conditions--not bad given the little bit clarity added with the cost of a little bit redundancy.

File: src/EnergyPlus/CondenserLoopTowers.cc [hunk 2]
```diff
@@ -3413,6 +3413,34 @@ namespace CondenserLoopTowers {
...
                 state, state.dataOutRptPredefined->pdchMechType, this->Name, DataPlant::PlantEquipTypeNames[static_cast<int>(this->TowerType)]);
             OutputReportPredefined::PreDefTableEntry(state, state.dataOutRptPredefined->pdchMechNomCap, this->Name, this->TowerNominalCapacity);
+
+            // create std 229 new table for cooling towers and fluid coolers
+            OutputReportPredefined::PreDefTableEntry(
+                state, state.dataOutRptPredefined->pdchCTFCType, this->Name, DataPlant::PlantEquipTypeNames[static_cast<int>(this->TowerType)]);
+            OutputReportPredefined::PreDefTableEntry(state,
+                                                     state.dataOutRptPredefined->pdchCTFCCondLoopName,
+                                                     this->Name,
+                                                     this->plantLoc.loopNum > 0 ? state.dataPlnt->PlantLoop(this->plantLoc.loopNum).Name : "N/A");
```
Comment (mjwitte) [inline src/EnergyPlus/CondenserLoopTowers.cc:3423]: This protection is fine, but is it necessary? When would a fluid cooler or tower not be connected to a plant loop? That seems possible only as an error condition. Or should "N/A" be "Not found"?

Comment (jcyuan2020) [reply src/EnergyPlus/CondenserLoopTowers.cc:3423]: Yes, this is a very generic protection---actually I remember I encountered many crashes in CI test files (or unit tests which might not be complete cases) with similar "plantloop cannot find smoothing and then crash" cases (not necessarily this type of cond loop for fluid coolers though). I cannot speak for sure in this specific scenario if this is going to be always good without the check. For the text, whatever suitable would be fine---I think for the other similar scenarios "N/A" was used---or maybe I can change the other places to "Not found".

**Comment by Myoldmopar** [comment]
I've got no issue here.  The extra protection is only happening early in the simulation, not all the time, so it shouldn't have an impact.  I will just do a quick build/test and get this in.

**Comment by Myoldmopar** [comment]
It's all fine here, dropping this one in now.  Thanks @jcyuan2020.  @mjwitte if you would like some adjustments to the protections added, we can do it in a follow-up.

**Pull request merged**